components/raftstore-v2/src/operation/query/follower_read_index.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_traits::{KvEngine, RaftEngine};
+use kvproto::raft_cmdpb::{self, CmdType, RaftCmdRequest, RaftCmdResponse};
+use pd_client::INVALID_ID;
+use raftstore::{
+    store::{
+        cmd_resp,
+        metrics::RAFT_READ_INDEX_PENDING_COUNT,
+        msg::{ErrorCallback, ReadCallback},
+        peer::propose_read_index,
+        Transport,
+    },
+    Error,
+};
+use slog::{debug, error, info, o, Logger};
+use tikv_util::{box_err, time::monotonic_raw_now};
+use time::Timespec;
+
+use crate::{
+    batch::StoreContext,
+    raft::Peer,
+    router::{message::RaftRequest, response_channel::QueryResChannel},
+    Result,
+};
+impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
+    /// read index on follower
+    ///
+    /// return true if it's proposed.
+    pub(crate) fn follower_read_index<T: Transport>(
+        &mut self,
+        poll_ctx: &mut StoreContext<EK, ER, T>,
+        mut req: RaftCmdRequest,
+        mut err_resp: RaftCmdResponse,
+        ch: QueryResChannel,
+        now: Timespec,
+    ) -> bool {
+        if !self.is_leader() && self.leader_id() == INVALID_ID {
```

Comment:
!self.is_leader() may be redundant here.

Reply:
good catch.